src/js/react/useSyncExternalStoreWithSelector.ts

```diff
@@ -0,0 +1,143 @@
+/**
```

Comment:
Per the code comment, this is a port (mostly copy) of the official/published https://github.com/facebook/react/blob/main/packages/use-sync-external-store/src/useSyncExternalStoreWithSelector.js from Facebook/Meta. By porting this (which is highly unlikely to change), we sidestep adding additional dependencies to Media Chrome that are only needed for the react-specific module. This is unlike our general `react` module dependency, as we can assume folks using our react code already also have a dependency on react in their application, library, or the like.